lib/middleware/anonymous_cache.rb

```diff
@@ -65,13 +65,14 @@ def is_crawler?
       end
 
       def cache_key
-        @cache_key ||= "ANON_CACHE_#{@env["HTTP_ACCEPT"]}_#{@env["HTTP_HOST"]}#{@env["REQUEST_URI"]}|m=#{is_mobile?}|c=#{is_crawler?}|b=#{has_brotli?}|t=#{theme_key}"
+        @cache_key ||= "ANON_CACHE_#{@env["HTTP_ACCEPT"]}_#{@env["HTTP_HOST"]}#{@env["REQUEST_URI"]}|m=#{is_mobile?}|c=#{is_crawler?}|b=#{has_brotli?}|t=#{theme_id}"
       end
 
-      def theme_key
-        key, _ = @request.cookies['theme_key']&.split(',')
-        if key && Guardian.new.allow_theme?(key)
-          key
+      def theme_id
+        ids, _ = @request.cookies['theme_ids']&.split('|')
+        id = ids&.split(",")&.map(&:to_i)&.first
+        if id && Guardian.new.allow_theme?(id)
+          id
```

Comment:
we should prep guardian here to call allow theme on every id, not only the first.